File: archives/C#/0129. Sum Root to Leaf Numbers.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int SumNumbers(TreeNode root) {
        if(root==null)
            return 0;
        int sum=0;
        var rootQueue=new Queue<TreeNode>();
        rootQueue.Enqueue(root);
        var rootValQueue=new Queue<int>();
        rootValQueue.Enqueue(root.val);
        while(rootQueue.Count()!=0){
            int lenOfRootQueue=rootQueue.Count();
            for(int i=0;i<lenOfRootQueue;i++){
                TreeNode curRoot=rootQueue.Dequeue();
                int curVal=rootValQueue.Dequeue();
                if(curRoot.left==null && curRoot.right==null)
                    sum+=curVal;
                if(curRoot.left!=null){
                    rootQueue.Enqueue(curRoot.left);
                    rootValQueue.Enqueue(curVal*10+curRoot.left.val);
                }
                if(curRoot.right!=null){
                    rootQueue.Enqueue(curRoot.right);
                    rootValQueue.Enqueue(curVal*10+curRoot.right.val);
                }
            }
        }
        return sum;
    }
}

File: archives/20190519python/0129. Sum Root to Leaf Numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumNumbers(self, root: TreeNode) -> int:
        return self.sumOfNumbers(root,0)
 
    def sumOfNumbers(self,root,res):
        if not root:
            return 0
        if not root.left and not root.right:
            return res*10+root.val
        return self.sumOfNumbers(root.left,10*res+root.val)+self.sumOfNumbers(root.right,10*res+root.val)
